### PR TITLE
Installing detox + implementing UI tests

### PR DESCRIPTION
**Pull request by marvestt**
- Set up Detox for Android
- Fixed previous exceptions being thrown at Sign Up and Log In page
- Wrote an exhaustive list of UI tests for different possible Sign up cases
- UI tests for each edge case from Log In was written as well
- Attempted to implement integration tests, but was eventually removed due to errors
- Verifying various components are visible/not visible at different stages of UI testing

### CODE REVIEW

**Review by sha256rma — COMMENTED** [review]
Nothing much changed. added detox for e2e testings and added more tests

File: __tests__/Feed-test.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/** NOT AN ACTUAL TEST. DELETING FILE GIVES ERRORS WITH NODE MODULES. */
```
Comment (denny999222) [inline __tests__/Feed-test.js:1]: clarify or remove comment bc it can be confusing
